packages/eslint-plugin-ecmascript-compat/lib/rule.js

```diff
@@ -44,12 +41,35 @@ module.exports = {
               ],
             },
           },
+          overrideBrowserslist: {
+            oneOf: [{ type: 'string' }, { type: 'array', items: { type: 'string' } }],
+          },
+          browserslistOptions: {
+            type: 'object',
+            properties: {
+              config: {
+                type: 'string',
+              },
+              path: {
+                type: 'string',
+              },
+              env: {
+                type: 'string',
+              },
+            },
+            additionalProperties: false,
+          },
         },
         additionalProperties: false,
       },
     ],
   },
   create(context) {
+    const overrideBrowserslist = context.options?.[0]?.overrideBrowserslist;
+    const browserslistOptions = context.options?.[0]?.browserslistOptions;
+    const targets = targetRuntimes(overrideBrowserslist, browserslistOptions);
```

Comment:
I just checked, I see browserslist search-read-parse-etc. is cached, so there's no issue calling this for every file even in a huge codebase like yours.